### PR TITLE
Yanntwd spd logeucl metric

### PR DESCRIPTION
**Pull request by ythanwerdas**

### CODE REVIEW

File: geomstats/geometry/spd_matrices_space.py [hunk 1]
```diff
@@ -717,3 +717,145 @@ def exp_domain(self, tangent_vec, base_point):
...
+    def __init__(self, n):
+        dimension = int(n * (n + 1) / 2)
+        super(SPDMetricLogEuclidean, self).__init__(
+            dimension=dimension,
+            signature=(dimension, 0, 0))
+        self.n = n
+        self.space = SPDMatricesSpace(n)
+
+    def inner_product(self, tangent_vec_a, tangent_vec_b, base_point):
+        """
```
Comment (ninamiolane) [inline geomstats/geometry/spd_matrices_space.py]: Fix docstring format:
"""Compute the inner...

(See template shared by @cshewmake2 , this allows to generate the documentation website automatically).

File: geomstats/geometry/spd_matrices_space.py [hunk 2]
```diff
@@ -717,3 +717,145 @@ def exp_domain(self, tangent_vec, base_point):
...
+                                                            base_point)
+        product = gs.matmul(modified_tangent_vec_a, modified_tangent_vec_b)
+        inner_product = gs.trace(product, axis1=1, axis2=2)
+
+        inner_product = gs.to_ndarray(inner_product, to_ndim=2, axis=1)
+
+        return inner_product
+
+    def exp(self, tangent_vec, base_point):
+        """
```
Comment (ninamiolane) [inline geomstats/geometry/spd_matrices_space.py]: Same

File: geomstats/geometry/spd_matrices_space.py [hunk 3]
```diff
@@ -717,3 +717,145 @@ def exp_domain(self, tangent_vec, base_point):
...
+        log_base_point = gs.linalg.logm(base_point)
+        dlog_tangent_vec = self.space.differential_log(tangent_vec, base_point)
+        exp = gs.linalg.expm(log_base_point + dlog_tangent_vec)
+
+        if ndim == 2:
+            return exp[0]
+        return exp
+
+    def log(self, point, base_point):
+        """
```
Comment (ninamiolane) [inline geomstats/geometry/spd_matrices_space.py]: Same

File: geomstats/geometry/spd_matrices_space.py [hunk 4]
```diff
@@ -717,3 +717,145 @@ def exp_domain(self, tangent_vec, base_point):
...
+        log_point = gs.linalg.logm(point)
+        log = self.space.differential_exp(log_point - log_base_point,
+                                          log_base_point)
+
+        if ndim == 2:
+            return log[0]
+        return log
+
+    def geodesic(self, initial_point, initial_tangent_vec):
+        """
```
Comment (ninamiolane) [inline geomstats/geometry/spd_matrices_space.py]: Same

**Comment by ninamiolane** [comment]
Good to go when flake8 passes! (it is testing on geomstats, tests and examples now).

**Comment by ythanwerdas** [comment]
There are still many docstrings that don't fit the new convention. Do we have to wait until all are fixed to (re)submit PRs?

**Comment by stefanheyder** [comment]
@ythanwerdas I'd say yes, new code should always adhere to the current standards. You won't have to resubmit, it will be enough to push the changes to your repository and github will take care of updating the PR.

**Comment by ythanwerdas** [comment]
Thanks @stefanheyder 
Yes sure, but why did we accept the PR requiring the new docstring policy before rewriting all the docstrings? Now all the PRs will fail in Travis until this is fixed... no?

**Comment by cshewmake2** [comment]
Hey @ythanwerdas. 

First, you're right; we're still updating all of the old docstrings. I'll have several PRs on that today. 

Second, Travis should not fail with incorrect docstring syntax at present. This will only happen once we add flake8-docstring to the dev requirements file. 

This is so that we can transition the codebase over without causing Travis to fail. 

If you have any questions on docstring format, you can contact me directly; I'll have a document up linked by Contributing.md with this information soon.

**Comment by ninamiolane** [comment]
Seconding @cshewmake2 there: yes, we did not merge the PR #319 enforcing docstring format yet, the error you have on travis is an import order error, not a documentation error @ythanwerdas .

Yet, since you installed flake8-docstrings, you see the D documentation errors on your computer. You can temporarily ignore them with:

http://flake8.pycqa.org/en/3.1.1/user/ignoring-errors.html

**Comment by ythanwerdas** [comment]
Great! Many thanks @cshewmake2 @ninamiolane @stefanheyder, sorry for misunderstanding.
Can we help you on updating the docstrings @cshewmake2 ?

**Pull request merged**